packages/gatsby-source-contentful/src/source-nodes.js

```diff
@@ -466,61 +493,92 @@ export async function sourceNodes(
   for (let i = 0; i < contentTypeItems.length; i++) {
     const contentTypeItem = contentTypeItems[i]
 
-    if (entryList[i].length) {
-      reporter.info(
-        `Creating ${entryList[i].length} Contentful ${
-          pluginConfig.get(`useNameForId`)
-            ? contentTypeItem.name
-            : contentTypeItem.sys.id
-        } nodes`
-      )
+    const timer =
+      entryList[i].length > 0
+        ? reporter.createProgress(
+            `Creating Contentful ${
+              pluginConfig.get(`useNameForId`)
+                ? contentTypeItem.name
+                : contentTypeItem.sys.id
+            } nodes`
+          )
+        : null
+
+    if (timer) {
+      timer.start()
     }
 
+    let nodeCreatedCount = 0
     // A contentType can hold lots of entries which create nodes
     // We wait until all nodes are created and processed until we handle the next one
-    // TODO add batching in gatsby-core
-    await Promise.all(
-      createNodesForContentType({
-        contentTypeItem,
-        restrictedNodeFields,
-        conflictFieldPrefix,
-        entries: entryList[i],
+    await createNodesForContentType({
+      contentTypeItem,
+      restrictedNodeFields,
+      conflictFieldPrefix,
+      entries: entryList[i],
+      createNode: timer?.tick
```

Comment:
Can you please explain to me what the timer here does?

Reply:
It shows a progress bar with info about how many nodes are being created per second. I can change this part back but it was useful while debugging this large site

Reply:
no a progress bar is fine. I guess in environments like the CI it will still display the old logging style?